cfme/common/provider.py

```diff
@@ -130,6 +130,14 @@ def __init__(self, **kwargs):
             if self.type == 'service_account':
                 self.service_account = kwargs['service_account']
 
+    def __hash__(self):
+        return hash(self.key) ^ hash(type(self))
```

Comment:
both eq and __hash__ can be implemented in terms of  the tuple (type(self), self.key)